src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -3894,7 +3920,10 @@ void ApplicationManagerImpl::CloseNaviApp() {
     UnregisterApplication(app_id, ABORTED);
   }
 
-  navi_app_to_stop_.pop_front();
+  {
+    sync_primitives::AutoLock lock(navi_app_to_stop_lock_);
+    navi_app_to_stop_.pop_front();
+  }
```

Comment:
I think this should stay at the top of the function, otherwise the timer could theoretically fire for 2 different apps, but the same app ID would be read twice

Reply:
Ah, I see that this would cause issues, perhaps you could put this value to the end of the queue and then remove it from there after?

Reply:
This works, but the changes are not obvious, since this is a regression after https://github.com/smartdevicelink/sdl_core/pull/2425 I will try to rework that fix, and do not call EndService inside the UnregisterApp. I'll let you know when the changes are ready

Reply:
@jacobkeeler I forced the commits to make the good commit history. I reworked the fix from #2425 in 898b824f78954871c0a31f5f6dea7f095827f959 and now SDL works as expected in accordance with https://github.com/smartdevicelink/sdl_core/issues/1409 and https://github.com/smartdevicelink/sdl_core/issues/3328
Now I'm checking the regression and I will let you know when it will be passed.